### ISSUE TITLE
Video messages do not show thumbnails.

### ISSUE THREAD

**Issue by VincentSit** [jessesquires/JSQMessagesViewController#628]
Looks like `JSQVideoMediaItem` does not  automatically generate thumbnail.
How can I make it show thumbnails?

**Comment by relatedcode** [comment]
@jcoleman A possible implementation:

```
#import <AVFoundation/AVFoundation.h>

- (UIImage *)videoThumbnail:(NSURL *)video
{
	AVURLAsset *asset = [[AVURLAsset alloc] initWithURL:video options:nil];
	AVAssetImageGenerator *generator = [[AVAssetImageGenerator alloc] initWithAsset:asset];
	generator.appliesPreferredTrackTransform = YES;
	CMTime time = [asset duration];
	time.value = 0;
	
	NSError *error = nil;
	CMTime actualTime;
	
	CGImageRef image = [generator copyCGImageAtTime:time actualTime:&actualTime error:&error];
	UIImage *thumbnail = [[UIImage alloc] initWithCGImage:image];
	CGImageRelease(image);

	return thumbnail;
}
```

**Comment by relatedcode** [comment]
@jessesquires As far as I can see, the main problem here is the timing. It's too slow to download the video and generate thumbnail images on the fly.

My suggestion is:

1., generate the video thumbnail image before uploading/sending the video,
2., upload/send the thumbnail image with the video file,
3., display the thumbnail image regardless the video file.

To make this happen JSQVideoMediaItem should store/manage the thumbnail image as well.

A possible solution can be found here:

[VideoMediaItem.h](https://github.com/relatedcode/RealtimeChat/blob/master/RealtimeChat/Classes/Chat/MediaItems/VideoMediaItem.h)
[VideoMediaItem.m](https://github.com/relatedcode/RealtimeChat/blob/master/RealtimeChat/Classes/Chat/MediaItems/VideoMediaItem.m)
[ChatView.m](https://github.com/relatedcode/RealtimeChat/blob/master/RealtimeChat/Classes/Chat/ChatView.m)

**Comment by jessesquires** [comment]
Yep - my plan for implementing this would be to provide the tools/hooks necessary to achieve this, but leave it up to clients to decide exact implementations (like the flow you've described above)

**Comment by raghunath44** [comment]
How to play a video???

**Comment by jessesquires** [comment]
Completed in #1823

**Comment by Premal-Khetani** [comment]
Great Work guys. But when i install it using cocoapods then this file is missing :( what to do with it?

**Comment by apwelsh** [comment]
Is this even part if the 7.x branch?  I understood the issue to be addressed in 8.0.
